src/mem/metaslab.h

```diff
@@ -60,6 +60,12 @@ namespace snmalloc
      */
     bool sleeping_ = false;
 
+    /**
+     * Flag to indicate this is actually a large allocation rather than a slab
+     * of small allocations.
+     */
```

Comment:
I am going to say some things that are probably false, but a corrected version should probably end up somewhere in `docs/`.  I feel like I have written this before but I cannot find it.

Address space is in one of five states:
* Not ours, in which the `Pagemap` is entirely zero and the address space is not tracked by any `AddressSpaceCore`.
* Ours and in an `AddressSpaceCore`: the `Pagemap` `Metaslab*` points to a `FreeChunk`.  The `Pagemap` `remote` and `sizeclass` fields are both zero.
* Ours and in a local allocator's stacks as a large allocation: the Pagemap Metaslab* points to a `ChunkRecord`.  The Pagemap size class is a *large* size class and the remote pointer points at the owning allocator.
* Ours and allocated as a large allocation: the `Pagemap` `Metaslab*` points to a `Metaslab` whose `needed` is `1` to force us off the fast path and whose `large_` flag is `true`.  The `Pagemap` size class is a *large* size class and the remote pointer points at the owning allocator.
* Ours and in use as a slab of some flavor: the `Pagemap` `Metaslab*` points to a `Metaslab` whose `large_` flag is `false`.  The `Pagemap` size class is a *small* size class and the remote pointer points at the owning allocator.

Is all that right?  If it is, should we make `FreeChunk` also have a `MetaCommon` prefix like the others?

Reply:
Yeah, that is right. #361 is where you started documenting this.